src/providers/http_provider.py

```diff
@@ -54,11 +54,16 @@ def _get(
         Returns (data, meta)
         """
         with self.PROMETHEUS_HISTOGRAM.time() as t:
-            response = self.session.get(
-                self._urljoin(self.host, endpoint.format(*path_params) if path_params else endpoint),
-                params=query_params,
-                timeout=self.REQUEST_TIMEOUT,
-            )
+            try:
+                response = self.session.get(
+                    self._urljoin(self.host, endpoint.format(*path_params) if path_params else endpoint),
+                    params=query_params,
+                    timeout=self.REQUEST_TIMEOUT,
+                )
+            except Timeout as error:
```

Comment:
We need to add handle for this exception to BaseModule

Reply:
common `TimeoutError` re-raised from here and will be handled in base module